packages/workers/spaces/src/handlers/createSpace.ts

```diff
@@ -0,0 +1,108 @@
+import { Errors } from '@lenster/data/errors';
+import { Regex } from '@lenster/data/regex';
+import validateLensAccount from '@lenster/lib/validateLensAccount';
+import jwt from '@tsndr/cloudflare-worker-jwt';
+import type { IRequest } from 'itty-router';
+import { error } from 'itty-router';
+import { boolean, object, string } from 'zod';
+
+import type { Env } from '../types';
+
+type ExtensionRequest = {
+  accessToken: string;
+  isMainnet: boolean;
+  isTokenGated: boolean;
+  conditionType?: string;
+  conditionValue?: string;
+  startTime?: string;
+};
+
+type CreateRoomResponse = {
+  message: string;
+  data: { roomId: string };
+};
+
+const validationSchema = object({
+  accessToken: string().regex(Regex.accessToken),
+  isMainnet: boolean()
+});
+
+export default async (request: IRequest, env: Env) => {
+  const body = await request.json();
+  if (!body) {
+    return error(400, 'Bad request!');
+  }
+
+  const validation = validationSchema.safeParse(body);
+
+  if (!validation.success) {
+    return new Response(
+      JSON.stringify({ success: false, error: validation.error.issues })
+    );
+  }
+
+  const {
+    accessToken,
+    isMainnet,
+    isTokenGated,
+    conditionType,
+    conditionValue,
+    startTime
+  } = body as ExtensionRequest;
+
+  try {
+    const isAuthenticated = await validateLensAccount(accessToken, isMainnet);
+    if (!isAuthenticated) {
+      return new Response(
+        JSON.stringify({ success: false, error: Errors.SignWallet })
+      );
+    }
+    const { payload } = jwt.decode(accessToken);
+    let response;
+    if (isTokenGated) {
```

Comment:
We don't want to have condition here, instead add in body using spread op for condition

```
...(condition && {obj})
```